run/test/specs/disappearing_message_checks_android.spec.ts

```diff
@@ -0,0 +1,477 @@
+import {
+  DMTimeOption,
+  DisappearActions,
+  DisappearModes,
+} from "../../types/testing";
+import { sleepFor } from "./utils";
+import { newUser } from "./utils/create_account";
+import { newContact } from "./utils/create_contact";
+import { joinCommunity } from "./utils/join_community";
+import {
+  SupportedPlatformsType,
+  closeApp,
+  openAppTwoDevices,
+} from "./utils/open_app";
+import { setDisappearingMessage } from "./utils/set_disappearing_messages";
+
+async function disappearingImageMessage(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  const testMessage = "Testing disappearing messages for images";
+  const time = "1 minute";
+  const mode: DisappearActions = "sent";
+  // Create user A and user B
+  const [userA, userB] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+  ]);
+  await newContact(platform, device1, userA, device2, userB);
+
+  await setDisappearingMessage(
+    platform,
+    device1,
+    ["1:1", "Disappear after send option", "1 minute"],
+    device2
+  );
+  // TODO FIX CONTROL MESSAGES ON ANDROID
+  // await device2.disappearingControlMessage(
+  //   `${userA.userName} has set messages to disappear ${time} after they have been ${mode}.`
+  // ),
+  //   await device2.disappearingControlMessage(
+  //     `You set messages to disappear ${time} after they have been ${mode}.`
+  //   );
+  // Wait for control messages to disappear before sending image (to check if the control messages are interfering with finding the untrusted attachment message)
+  await sleepFor(60000);
+  await device1.sendImage(platform, testMessage);
+  await device2.clickOnElementAll({
+    strategy: "accessibility id",
+    selector: "Untrusted attachment message",
+  });
+  await device2.clickOnElementAll({
+    strategy: "accessibility id",
+    selector: "Download media",
+  });
+  await sleepFor(60000);
+
+  await Promise.all([
+    device1.hasElementBeenDeletedNew({
+      strategy: "accessibility id",
+      selector: "Message body",
+      maxWait: 1000,
+      text: testMessage,
+    }),
+    device2.hasElementBeenDeletedNew({
+      strategy: "accessibility id",
+      selector: "Message body",
+      maxWait: 1000,
+      text: testMessage,
+    }),
+  ]);
+  await closeApp(device1, device2);
+}
+
+async function disappearingVideoMessage(platform: SupportedPlatformsType) {
+  const { device1, device2 } = await openAppTwoDevices(platform);
+  // Create user A and user B
+  const [userA, userB] = await Promise.all([
+    newUser(device1, "Alice", platform),
+    newUser(device2, "Bob", platform),
+  ]);
+  const time: DMTimeOption = "1 minute";
+  const mode: DisappearActions = "sent";
+  await newContact(platform, device1, userA, device2, userB);
+
+  await setDisappearingMessage(
+    platform,
+    device1,
+    ["1:1", "Disappear after send option", "1 minute"],
```

Comment:
should this be reusing the var time/mode?

Reply:
yes.... technically
I'll update the time, i'm just struggling with the mode, there's alot of annoying nuances to 'send'/'sent' or 'read'. Will explain in person haha